docs/notebook_tutorial/ocr.ipynb

```diff
@@ -0,0 +1,535 @@
+    "from PIL import Image\n",
+    "import requests\n",
+    "import pytesseract"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### OCR Reader\n",
+    "\n",
+    "It's essentially a reader that reads image from an url, and load the image into `ImagePayload` in the DataPack."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class OcrReader(PackReader):\n",
+    "    r\"\"\"\n",
+    "    :class:`OcrReader` is designed to read an image\n",
+    "    \"\"\"\n",
+    "    def _collect(self, image_path) -> Iterator[Any]:  # type: ignore\n",
+    "        r\"\"\"Should be called with param ``Image_directory`` which is a path to a\n",
+    "        folder containing Image files.\n",
+    "\n",
+    "        Args:\n",
+    "            Image_directory: Image directory containing the files.\n",
+    "\n",
+    "        Returns: Iterator over paths to Image files\n",
+    "        \"\"\"\n",
+    "        # construct ImageMeta and store it in DataPack\n",
+    "        return image_path\n",
+    "\n",
+    "    def _parse_pack(self, data_source: str) -> Iterator[DataPack]:\n",
+    "        r\"\"\"Takes a raw string and converts into a :class:`~forte.data.data_pack.DataPack`.\n",
+    "\n",
+    "        Args:\n",
+    "            data_source: str that contains text of a document.\n",
+    "\n",
+    "        Returns: :class:`~forte.data.data_pack.DataPack` containing Document.\n",
+    "        \"\"\"\n",
+    "        pack: DataPack = DataPack()\n",
+    "        payload_idx = 0\n",
+    "        \n",
+    "        # Read in Image data and store in DataPack\n",
+    "        # add Image payload into DataPack.payloads\n",
+    "        ip = ImagePayload(pack, payload_idx)\n",
+    "        image_data = Image.open(requests.get(data_source, stream=True).raw)\n",
+    "        ip.set_cache(image_data)\n",
+    "        yield pack"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## OCR Processor\n",
+    "It analyzes the image in the DataPack ImagePayload, and predict the bounding box for each character. It wraps predicted bounding boxes and characters in forte ontology and write back into the DataPack."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class OcrProcessor(PackProcessor):\n",
+    "    \"\"\"\n",
+    "    An audio processor for speaker segmentation.\n",
+    "    \"\"\"\n",
+    "    def initialize(self, resources: Resources, configs: Config):\n",
+    "        super().initialize(resources, configs)\n",
+    "\n",
+    "    def _process(self, input_pack: DataPack):\n",
+    "        image_data = input_pack.image_payloads[0].cache\n",
+    "        ocr_text = pytesseract.image_to_string(image_data)\n",
+    "        input_pack.set_text(ocr_text)\n",
```

Comment:
In the processor, a few inline comment explaining what we are doing would be helpful

Reply:
Added.